src/main/java/com/google/sps/data/CalendarUtility.java

```diff
@@ -0,0 +1,98 @@
+package com.google.sps.data;
+
+import com.google.api.client.auth.oauth2.Credential;
+import com.google.api.client.extensions.appengine.datastore.AppEngineDataStoreFactory;
+import com.google.api.client.extensions.appengine.http.UrlFetchTransport;
+import com.google.api.client.googleapis.auth.oauth2.GoogleAuthorizationCodeFlow;
+import com.google.api.client.auth.oauth2.TokenResponse;
+import com.google.api.client.googleapis.auth.oauth2.GoogleClientSecrets;
+import com.google.api.client.http.GenericUrl;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.client.util.Preconditions;
+import com.google.api.client.util.store.DataStoreFactory;
+import com.google.api.services.calendar.Calendar;
+import com.google.api.services.calendar.CalendarScopes;
+import com.google.appengine.api.users.UserServiceFactory;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.Collections;
+
+import javax.servlet.http.HttpServletRequest;
+
+/**
+ * Utility class for Google OAuth flow helpers.
+*/
+public final class CalendarUtility {
+  /**
+   * Global instance of the {@link DataStoreFactory}. The best practice is to make it a single
+   * globally shared instance across your application.
+  */
+  private static final AppEngineDataStoreFactory DATA_STORE_FACTORY =
+    AppEngineDataStoreFactory.getDefaultInstance();
+  
+  /** Global instance of the HTTP transport. */
+  static final HttpTransport HTTP_TRANSPORT = new UrlFetchTransport();
```

Comment:
Is there a reason for HTTP_TRANSPORT and JSON_FACTORY to be not private?

Reply:
You are right. I will add it in the next pull request.